Fix Telegraf repo URL handling for RedHat hosts

trauta opened the pull request:
Changes:
- Use `${telegraf::repo_location}` in baseurl
- Use [new baseurl scheme](https://docs.influxdata.com/telegraf/v1/install/?t=RedHat+%26amp%3B+CentOS#download-and-install-telegraf) for default RedHat hosts

--
This PR is based on the work by @alexeiser and his open PR: https://github.com/voxpupuli/puppet-telegraf/pull/250
Since I need the fixed usage of `${telegraf::repo_location}` rather quickly, I'll open this PR based on his works with the updated baseurl scheme.

alexeiser commented on manifests/install.pp:
```diff
@@ -108,9 +108,16 @@
     'RedHat': {
       if $telegraf::manage_repo {
         if $facts['os']['name'] == 'Amazon' {
```
Assuming the new repo works for AmazonLinux2/2023 i think the if block cab be removed.

trauta replied:
Yeah, you're right, thanks for the hint! Did a small test with Amazon Linux 2023.

trauta committed:
Fix Telegraf repo URL handling for RedHat hosts

- Use `${telegraf::repo_location}` in baseurl
- Use new baseurl scheme [1] for default RedHat hosts
- Update `telegraf_spec.rb` to use new baseurl scheme

[1]
https://docs.influxdata.com/telegraf/v1/install/?t=RedHat+%26amp%3B+CentOS#download-and-install-telegraf
